src/views/Settings.vue

```diff
@@ -68,7 +68,7 @@ export default defineComponent({
     setFacility (facility: any) {
       if(this.userProfile && this.userProfile.facilities) {
         this.userProfile.facilities.map((fac: any) => {
-          if (fac.facilityId == facility['detail'].value) {
+          if (fac.facilityId == facility['detail'].value && this.currentFacility.facilityId != facility['detail'].value) {
```

Comment:
Add this check before the userProfile check as if the values are not equal then there is no need to loop over user facilities.
Also add a comment Todo that why we have added this check.

Reply:
Added the check before userProfile check and comment.